### ISSUE TITLE
Select folders/files while adding torrents

### ISSUE THREAD

**Issue by SumatraPeter** [devgianlu/Aria2App#160]
As the subject states, it would be great if the app displayed all the folders/files in a torrent when one is selected (or downloaded if a URI is provided), with checkboxes against each one to select/deselect what to download _before_ the torrent is added to the download list.

Also, although there's a "Pause download when added" option for each individual download, a global option to always add downloads (esp. torrents) in a paused state would be nice to have as well.

**Comment by devgianlu** [comment]
I don't think this is possible since we don't know the files that are included in the torrent before downloading the metadata, aria2 doesn't provide this option either.

To generally pause all downloads when starting you can set `pause` to `true` in the global settings.

**Comment by SumatraPeter** [comment]
> we don't know the files that are included in the torrent before downloading the metadata

Not sure I quite understand. Irrespective of whether an HTTP(s) or magnet or any other link is provided, doesn't the actual .torrent file need to be downloaded first (from the swarm in case of magnet links) before its contents are downloaded? Moreover in the case of a local .torrent file being added, its contents can be immediately read by the app and a file selection dialog displayed before the torrent is added to the download queue. If desktop torrent clients can do it, why not this app?

Thanks for the tip about global settings.

**Comment by masonh** [comment]
I haven't tried this in the app, but the aria2 option `pause-metadata` might at least solve the issue of getting metadata before pausing.

**Comment by devgianlu** [comment]
https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-select-file is used to tell aria2 which files to include in the download, I need to figure out if they indexes are related in any way to the ones of the torrent metadata.
